File: corpus/cases/verified/missing-schema-validation/clean-fastapi-pydantic-model/tree/app/routes/invoices.py
from __future__ import annotations

from fastapi import APIRouter, Depends

from ..db import Database, get_db
from ..schemas import CreateInvoice, InvoiceCreated

router = APIRouter(prefix="/v1", tags=["invoices"])


@router.post("/invoices", response_model=InvoiceCreated, status_code=201)
async def create_invoice(
    body: CreateInvoice,
    db: Database = Depends(get_db),
) -> InvoiceCreated:
    row = await db.fetchrow(
        "insert into invoices (customer_id, amount_cents, currency, memo) "
        "values ($1, $2, $3, $4) returning id, amount_cents, currency",
        body.customer_id,
        body.amount_cents,
        body.currency,
        body.memo,
    )
    return InvoiceCreated(
        id=row["id"],
        amount_cents=row["amount_cents"],
        currency=row["currency"],
    )

File: corpus/cases/verified/missing-schema-validation/clean-validated-in-middleware/tree/src/routes/invoices.ts
import { Router } from 'express';
import { validateBody } from '../middleware/validate.js';
import { CreateInvoice, ListInvoices } from '../schemas/invoice.js';
import { pool } from '../db.js';

export const invoiceRouter = Router();

invoiceRouter.post('/invoices', validateBody(CreateInvoice), async (req, res) => {
  const { customerId, amountCents, currency, memo } = req.body;

  const { rows } = await pool.query(
    'insert into invoices (customer_id, amount_cents, currency, memo) ' +
      'values ($1, $2, $3, $4) returning id',
    [customerId, amountCents, currency, memo],
  );

  res.status(201).json({ id: rows[0].id });
});

invoiceRouter.post('/invoices/search', validateBody(ListInvoices), async (req, res) => {
  const { status, take } = req.body;

  const { rows } = await pool.query(
    'select id, amount_cents, currency from invoices where status = $1 limit $2',
    [status, take],
  );

  res.json({ invoices: rows });
});
